File: 14-10-2025_Sum_of_Nodes_in_BST_Range.py
"""
class Node:
    def __init__(self, val):
        self.data = val
        self.left = None
        self.right = None
"""
class Solution:
    def nodeSum(self, root, l, r):
        # code here
        s=0
        def recurse(root):
            nonlocal s
            if root!=None:
                if l<=root.data<=r:
                    s+=root.data
                recurse(root.left)
                recurse(root.right)
        recurse(root)
        return s
        
        

File: 14-10-2025_Sum_of_Nodes_in_BST_Range.java
/*
class Node {
    int data;
    Node left, right;
    Node(int val) {
        data = val;
        left = right = null;
    }
}
*/
class Solution {
    int s;

    int nodeSum(Node root, int l, int r) {
        s = 0;
        recurse(root, l, r);
        return s;
    }

    void recurse(Node root, int l, int r) {
        if (root != null) {
            if (l <= root.data && root.data <= r)
                s += root.data;
            recurse(root.left, l, r);
            recurse(root.right, l, r);
        }
    }
}
